common/src/main/scala/org/apache/celeborn/common/util/Utils.scala

```diff
@@ -414,8 +414,28 @@ object Utils extends Logging {
     customHostname = Some(hostname)
   }
 
+  val ipv6AddressPattern = """^\[.*:.*\]$""".r
```

Comment:
Inet6Address.getHostAddress will return an IPv6 string without the "[]". In this case, I think this check is redundant.

Reply:
I was thinking that in case `CELEBORN_LOCAL_IP` is set to ipv6 address with `[]`. Do you think we still don't need it? If we don't need it, then this code can jut become: 
```
  def localHostName(conf: CelebornConf): String = customHostname.getOrElse {
    if (conf.bindPreferIP) {
      localIpAddress match {
        case ipv6Address: Inet6Address => s"[${ipv6Address.getHostAddress}]"
        case other => other.getHostAddress
      }
    } else {
      localIpAddress.getCanonicalHostName
    }
  }
```
We do not need the other method `addBracketsIfNeeded` or any unit tests required. cc @mridulm

Reply:
According to WIKI(https://en.wikipedia.org/wiki/IPv6_address). An IPv6 address will need to be enclosed in square brackets if it's in network resource identifiers.  Set CELEBORN_LOCAL_IP is an IP address, it should be enclosed in square brackets. So I think this won't be a problem here.